### DISCUSSION TITLE
docs/Single-Responsibility-Principle-(SRP)

### DISCUSSION THREAD

**Discussion by giscus** [samuelgfeller/blog#2] [Comments]
# docs/Single-Responsibility-Principle-(SRP)

The Single Responsibility Principle (SRP) states that a class should have only one reason to change and only a single responsibility. This document explains the use-case-based approach and why it's beneficial.

https://samuel-gfeller.ch/docs/Single-Responsibility-Principle-(SRP)

**Comment by Alpine418** [comment]
Thank you for your inspiring work and documentation!
